subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/transform/ArtifactTransformBuildOperationIntegrationTest.groovy

```diff
@@ -741,4 +867,43 @@ class ArtifactTransformBuildOperationIntegrationTest extends AbstractIntegration
             acc + (acc.endsWith(":") && path.startsWith(":") ? path.substring(1) : path)
         }
     }
+
+    boolean matchTransformationIdentity(actual, TransformationIdentityWithoutId expected) {
+        actual.nodeType.toString() == ARTIFACT_TRANSFORM &&
```

Comment:
🤔 I wonder if we should separate the matching from asserting. When you assert that this method returns true, then Spock's power asserts can't look into here and you'd need to figure out from the arguments if the two objects are the same.

Reply:
It's not going to be super helpful out-of-the-box, because we do graph assertions before the execution assertions. In the graph assertions, we use matching to find expected nodes. If production code breaks the test, or you wrote the expected fields incorrectly in the test, then you would still get an opaque error that the node in the graph was not found. So you'll never get to the power assertions that would tell you what is wrong.

Having said that, adding an assertion-based method would be useful for debugging, as you could comment out the graph matching code in the test locally, and see what was actually different. So, I'll add it.

Reply:
Done